src/modules/dataFetching/components/GenericTableWithData.tsx

```diff
@@ -61,15 +55,10 @@ export interface Props<
     rows: RowDataType[],
     loading?: boolean
   ) => ColumnDef<RowDataType>[]; // dynamically define column defs based on current data
-  filters?:
-    | TableFilterType<FilterOptionsType>
-    | ((
-        baseFilters: TableFilterType<FilterOptionsType>
-      ) => TableFilterType<FilterOptionsType>);
-  filterPickListArgs?: PickListArgs;
+  filters?: TableFilterType<FilterOptionsType>;
   sortOptions?: SortOptionsType;
   defaultSortOption?: keyof SortOptionsType;
-  defaultFilters?: Partial<FilterOptionsType>;
+  defaultFilterValues?: Partial<FilterOptionsType>;
   showFilters?: boolean;
   noSort?: boolean;
   noFilter?: boolean;
```

Comment:
My big remaining question around this refactor is the `showFilters` / `noFilter` confusion. I think I just don't understand their meaning based on the names -- and there were a couple of components that had both `showFilters` and `noFilter`, which really confused me (ClientScanCards, ClientCaseNotes, and EnrollmentCaseNotes).

If these mean what they sound like, then I think the ideal API for this GenericTable component is that neither one of them exists, and instead the filters are either shown or hidden based on whether `filters` is passed in by the caller.

Reply:
Thanks for the clarification about `showFilters` on the comment thread above. I'm now not quite sure what the cleanest API would be -- maybe...
- `showFilters` -> renamed to `showTopToolbar`, and either
  - A) the top toolbar is always shown if `filters` are provided, but if not, you can *separately* specify that it should be shown anyway with `showTopToolbar`, OR,
  - B) you must always pass `showTopToolbar` in addition to `filters` if you want the top toolbar to be shown
  - Maybe A is preferable because it avoids the unexpected situation where you're trying to pass in `filters` and confused about why they aren't showing up. ?

- `noFilter` -> I could also go either way
  - A) remove this prop, decide whether or not to show filters based on whether `filters` are passed. OR
  - b) keep it as a way to override, regardless of whether `filters` are passed.
  - Looking at the code, it seems like this was probably invented to give the ability to override when GenericTable always attempted to automatically create filters for the given record type. Now that that's not happening anymore, it seems like `noFilter` prop is not needed, but maybe we should keep it for parity with `noSort` which is still needed and used in a handful of places.

Reply:
1)
Ahemmm yes a classic prop dilemma. I wonder if React pros have an opinion about this. I prefer going with option A because it feels more intuitive when using the component. But it might be a personal preference thing.

2)
IMO `noFilter` is no longer necessary, because `filters` are now passed explicitly. I believe this was added to handle scenarios where the table was auto-generating the filter list, and we wanted a prop to let us hide them. Now with this refactor, filters are _always_ passed explicitly, so we shouldn't need this. 
Callers that previously passed `noFilter` should just...not pass any filters :) much nicer.

Reply:
Ok! 1) Agree, choosing 1A. 2) Agree, removing `noFilter`